toolchain/driver/driver.cpp

```
@@ -209,6 +209,11 @@ auto Driver::RunDumpSubcommand(DiagnosticConsumer& consumer,
   }
   CARBON_VLOG() << "semantics_ir: " << semantics_ir;
 
+  // Unlike previous steps, errors block further progress.
+  if (has_errors) {
+    return false;
```

Maybe at least a `VLOG` here rather than relying on the absence of the one below?

---

Right now, we don't vlog when a step results in an error; are you wanting that to be changed in general? I'd probably do it on line 204 and similar, where the error is returned from a call, not here.

Note though, I've typically been vlogging things that aren't already printed. In the case of errors, there'll be diagnostic output indicating the error, and the tool should return an error.

---

I'm adding a VLOG per the request.

---

I was specifically hoping for a VLOG to indicate that at this point we have to give up. Technically, the lack of the IR dump vlog tells you that, but it seemed nice to be explicit that this is in fact the fatal point in the process.

Anyways, still LG and thanks!